### ISSUE TITLE
encrypted password bind

### ISSUE THREAD

**Issue by JCarlos12** [mcavage/node-ldapjs#410]
using the **client.bind function**

with an encrypted password results to a **StrongAuthRequiredError**

is this feature already available or is there another approach?

Thanks

**Comment by arekkusu** [comment]
I am unsure what you meant by "with an encrypted password". Do you simply mean you're using SSL/TLS (ldaps) ? In any case:

> The bind API only allows LDAP 'simple' binds (equivalent to HTTP Basic Authentication) for now.
> http://ldapjs.org/client.html

I believe **StrongAuthRequiredError** indicates the server does want to use something else than plaintext  (there are other option such as DIGEST-MD5 or GSS authentication)

Normally if you're connecting using SSL/TLS using 'simple' bind is fine as the password is encrypted in transit. You might want to check with the admin of the ldap server you're connecting to.

**Comment by arekkusu** [comment]
Yes that would be fine and I don't really see a better solution.
Important thing is to use a good password hash (for example SHA2) on the LDAP server side.

My understanding is that most (all?) system that transmit a hash instead of the actually password have a significant disadvantage: password must be store in clear text on the server side.
(that might be file for a shared random PSK but a bad idea for user password)

**Comment by jsumners** [comment]
⚠️ This issue has been locked due to age. If you have encountered a recent
problem that seems to be covered by this issue, please open a new issue.

Please include a [minimal reproducible example](https://en.wikipedia.org/wiki/Minimal_reproducible_example)
when opening a new issue.
